drivers/spi/opentitan/driver.h

```diff
@@ -0,0 +1,118 @@
+#include <stdint.h>
+#include <stdbool.h>
+
+typedef struct spi_driver_data {
+    // Per request state
+    spi_cs_line_t cs_line;
+    spi_cmd_t *cmd;
+    uint16_t num_cmds;
+    void *slice_base;
+    // Request in-progress state
+    uint8_t cmd_in_progress;
```

Comment:
If this is an index, is it intended that the cmd_in_progress is a smalelr type than num_cmds?